repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3050,38 +3049,41 @@ public void cleanUpClassificationPropagation(String classificationName, int batc
                 int offset = 0;
                 do {
                     try {
-                        int toIndex = Math.min((offset + CHUNK_SIZE_TEMP), currentAssetsBatchSize);
+                        int toIndex = Math.min((offset + CHUNK_SIZE), currentAssetsBatchSize);
                         List<AtlasVertex> entityVertices = currentAssetVerticesBatch.subList(offset, toIndex);
                         List<String> impactedGuids = entityVertices.stream().map(GraphHelper::getGuid).collect(Collectors.toList());
                         GraphTransactionInterceptor.lockObjectAndReleasePostCommit(impactedGuids);
-
                         for (AtlasVertex vertex : entityVertices) {
                             List<AtlasClassification> deletedClassifications = new ArrayList<>();
                             List<AtlasEdge> classificationEdges = GraphHelper.getClassificationEdges(vertex, null, classificationName);
                             classificationEdgeCount += classificationEdges.size();
-                            for (AtlasEdge edge : classificationEdges) {
-                                try {
-                                    AtlasClassification classification = entityRetriever.toAtlasClassification(edge.getInVertex());
-                                    deletedClassifications.add(classification);
-                                    deleteDelegate.getHandler().deleteEdgeReference(edge, TypeCategory.CLASSIFICATION, false, true, null, vertex);
-                                }
-                                catch (IllegalStateException | AtlasBaseException e){
-                                    e.printStackTrace();
+                            int batchSize = CHUNK_SIZE;
+                            for (int i = 0; i < classificationEdges.size(); i += batchSize) {
+                                int end = Math.min(i + batchSize, classificationEdges.size());
+                                List<AtlasEdge> batch = classificationEdges.subList(i, end);
+                                for (AtlasEdge edge : batch) {
+                                    try {
+                                        AtlasClassification classification = entityRetriever.toAtlasClassification(edge.getInVertex());
+                                        deletedClassifications.add(classification);
+                                        deleteDelegate.getHandler().deleteEdgeReference(edge, TypeCategory.CLASSIFICATION, false, true, null, vertex);
+                                    } catch (IllegalStateException | AtlasBaseException e) {
+                                        e.printStackTrace();
+                                    }
                                 }
```

Comment:
This for loop needs to be optimized
Only commit if edges size is 100
i.e. Imagine having 150 assets each having only single edge to this Tag, such a case will call transactionInterceptHelper.intercept(); 150 times where we only want to call it twice

Reply:
Also, mind sending notifications part where you need to call send notifications still 150 times in this case but only after interceptor call
So you can maintain a map which will hold all assets & classificationList removed in 100 batch & one commit is successful you iterate map to send 100 notifications

Reply:
Another scenario for notifications part would be single asset having more that 100 edges, surely edge deleting must be in 100 batch but notifications part should be only called at the end when you ensure all edges for the Asset are deleted